local-dev/src/components/App.js

```diff
@@ -0,0 +1,28 @@
+/*
+Component that wraps any child component requiring our Redux data store.
+*/
+
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import * as actionCreators from '../actions/actionCreators';
+import Main from './Main';
+
+// Converts state to an object to be passed as props to Main component.
+function mapStateToProps(state) {
+    return {
+        global: state.global,
+        serviceDetail: state.serviceDetail
+    };
+}
+
+// Allows all action creators (from actions/actionCreators.js) to be called
+// directly (instead of through the 'dispatch' method).
```

Comment:
Yes, good question. Redux gets a *little* more complicated when performing async actions.

Using `bindActionCreators` in `App.js` lets us call actions directly from components that are children of the `App` component - e.g. [calling the `clearDesktopNotify` action in `GlobalStatus.js`](https://github.com/mozmar/status.mozmar.org/pull/29/files#diff-16907810b53ab5535c8298943076b90eR72). It's really just a convenience. From the docs:

> The only use case for `bindActionCreators` is when you want to pass some action creators down to a component that isn't aware of Redux, and you don't want to pass `dispatch` or the Redux store to it.

We are calling the `dispatch` method directly in the async action creators (as in your link above) because we haven't used the `bindActionCreators` shortcut, as it wouldn't really be a convenience in that file/scenario. `actionCreators.js` is already aware of Redux - we don't need to explicitly pass it a reference to the store or to the `dispatch` method (as we would have to do with our custom components).

As I understand it (which is admittedly at an amateur level), `actionCreators.js` is part of the Redux piece of the app, whereas our components are part of the React piece. `bindActionCreators` provides a handy link between the two.